overlord/ifacestate/handlers.go

```diff
@@ -458,13 +444,9 @@ func (m *InterfaceManager) doDisconnect(task *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	affectedConns, err := m.repo.ResolveDisconnect(plugRef.Snap, plugRef.Name, slotRef.Snap, slotRef.Name)
```

Comment:
Can we please just ensure that it will error properly instead of arbitrarily? In other words, check that the refs look properly resolved and if not return an error such as "snapd changed, please retry the operation".